vault/identity_store_oidc_test.go

```diff
@@ -1056,9 +1056,9 @@ func TestOIDC_PeriodicFunc(t *testing.T) {
 			setNextSigningKey: true,
 			testCases: []testCase{
 				{1, 2, 2},
-				{2, 2, 4},
-				{3, 2, 4},
-				{4, 2, 4},
+				{2, 2, 3},
+				{3, 2, 2},
+				{4, 2, 3},
```

Comment:
Can you explain how the number of public keys is determined based on the cycle?

Reply:
Good question! So each key has a next, current, and prior (if inside its [`verification_ttl`](https://developer.hashicorp.com/vault/api-docs/secret/identity/tokens#verification_ttl)). Each cycle in the test rotates the key, resulting in the following changes:
- next -> current
- current -> prior
- prior -> deleted if `verification_ttl` is up

It appeared as if we were looking for exact counts but our [test condition](https://github.com/hashicorp/vault/blob/main/vault/identity_store_oidc_test.go#L1168) was expecting that we have _at least_ a certain amount of keys. I was able to determine the exact counts after changing the assertion to an equality. I've added comments explaining why they change through cycles in https://github.com/hashicorp/vault/pull/24332/commits/924cb42a4b504e11f54f08df8f5c907493288fd4. I also verified that the behavior is the same on main where we're still writing the default key (except counts are all +2 🙂)

Reply:
Thanks!